modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/near/GridNearTxLocal.java

```diff
@@ -4190,4 +4224,150 @@ public boolean addTimeoutHandler() {
             "mappings", mappings,
             "super", super.toString());
     }
+
+    /**
+     * Creates savepoint.
+     *
+     * @param name Savepoint ID.
+     * @param overwrite Overwrite savepoint flag.
+     */
+    @Override public void savepoint(String name, boolean overwrite) throws IgniteCheckedException {
+        txState.awaitLastFuture(cctx);
+
+        checkValid();
+
+        if (state() != ACTIVE) {
+            throw new IllegalStateException("Trying to create savepoint not in ACTIVE transaction." +
+                " Savepoint will not be created.");
+        }
+
+        ListIterator<TxSavepointLocal> spIter = findSP(name);
+
+        if (spIter != null) {
+            if (overwrite) {
+                releaseSP(spIter);
+            } else {
+                throw new IllegalArgumentException("Savepoint \"" + name + "\" already exist. " +
```

Comment:
as mentioned in ticket description -  savepoint("pointname") should erase old savepoint

Reply:
Yes, savepoint(String name) is a has proxy realisation and calls savepoint(name, false), which means every user's savepoint(String name) call will throw exception if such point exists.

Reply:
Ok, than change ticket description or code